src/Planner/PlannerJoinTree.cpp

```diff
@@ -1414,7 +1414,14 @@ std::tuple<QueryPlan, JoinPtr> buildJoinQueryPlan(
     trySetStorageInTableJoin(right_table_expression, table_join);
     auto prepared_join_storage = tryGetStorageInTableJoin(right_table_expression, planner_context);
     auto hash_table_stat_cache_key = preCalculateCacheKey(right_table_expression, select_query_info);
-    auto join_algorithm = chooseJoinAlgorithm(table_join, prepared_join_storage, left_header, right_header, planner_context->getQueryContext(), std::move(hash_table_stat_cache_key));
+    auto join_algorithm = chooseJoinAlgorithm(
+        table_join,
+        prepared_join_storage,
+        left_header,
+        right_header,
+        planner_context->getQueryContext(),
+        std::move(hash_table_stat_cache_key),
+        /*rhs_estimation=*/{});
```

Comment:
this means it will not take any effect if took this branch

https://github.com/ClickHouse/ClickHouse/blob/5daf76d4d8d8e104a74c0766fb3eac557c7c4c28/src/Planner/PlannerJoinTree.cpp#L2044-L2046